src/Assets/File.php

```diff
@@ -639,12 +640,65 @@ protected function onBeforeWrite()
             $this->OwnerID = Member::currentUserID();
         }
 
-        // Set default name
-        if (!$this->getField('Name')) {
-            $this->Name ="new-" . strtolower($this->class);
+        $name = $this->getField('Name');
+        $title = $this->getField('Title');
+
+        $changed = $this->isChanged('Name');
+
+        // Name can't be blank, default to Title
+        if (!$name) {
+            $changed = true;
+            $name = $title;
+        }
+
+        $filter = FileNameFilter::create();
+        if ($name) {
+            // Fix illegal characters
+            $name = $filter->filter($name);
+        } else {
+            // Default to file name
+            $changed = true;
+            $name = $this->i18n_singular_name();
```

Comment:
I’m using singular_name() rather than basename(__CLASS__), since that’s abusing basename() for FQCN stripping, and doesn’t work consistently (backslashes only stripped on windows?)